src/main/fc/fc_msp.c

```diff
@@ -2180,8 +2181,10 @@ static mspResult_e mspCommonProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
                     uint8_t x = (value & ((1 << 5) - 1));
                     uint8_t y = ((value >> 5) & ((1 << 5) - 1));
                     uint8_t visible = (value & 0x0800) ? 1 : 0;
-                    osdConfigMutable()->item_pos[addr][0] = x;
-                    osdConfigMutable()->item_pos[addr][1] = y;
+                    uint8_t origin = (value & 0xF000) >> 12;
+                    osdConfigMutable()->item_pos[addr].x = x;
```

Comment:
I know, if the idea of the anchoring is approved I will rewrite this code and the associated code in the configurator gui.